### PR TITLE
Add support for release candidates

### PR DESCRIPTION
**Pull request by mattolson**
#### What?
* In stencil-release, add support for release candidates. These
  have version strings with `-rc.{n}` at the end, and are meant
  for internal testing prior to a real version bump.
* For release candidates, parse the changelog for the release notes,
  but don't update the changelog file itself.

#### Screenshot

Pushing a cornerstone release candidate:
```
[matt.olson@SFO-MBP-OLSON:~/workspace/cornerstone] $ ../stencil-cli/bin/stencil-release
? What type of release would you like to do? Current version: 1.11.0-rc.2: Release Candidate:  1.11.0-rc.3   Internal release for testing.
? What git remote repository would you like to push the release to? origin: git@github.com:mattolson/cornerstone.git
? Create a github release and upload the bundle zip file? Yes
? Proceed? Yes
Validating theme...
Running ESLint...

SCSS Parsing Started...
Template Parsing Started...
Language Files Parsing Started...
Building Theme Schema File...
Theme task Started...
ok -- Language Files Parsing Finished
ok -- Theme Schema Building Finished
ok -- SCSS Parsing Finished
ok -- Template Parsing Finished
ok -- Theme task Finished
Generating Manifest Started...
ok -- Manifest Generation Finished
ok -- Zipping Files Finished
Pushing Changes to origin...
Creating Github Release...
Uploading Bundle File...
Release url: https://github.com/mattolson/cornerstone/releases/tag/1.11.0-rc.3
Bundle download url: https://github.com/mattolson/cornerstone/releases/download/1.11.0-rc.3/Cornerstone-1.11.0-rc.3.zip
done
```

### CODE REVIEW

File: lib/release/questions.js [hunk 1]
```diff
@@ -11,34 +11,44 @@ function askQuestions(themeConfig, githubToken, remotes, callback) {
...
+                value: 'custom',
             }],
-            filter: bumpType => semver.inc(currentVersion, bumpType),
         },
         {
             name: 'version',
             type: 'input',
             message: 'What specific version would you like',
-            when: answers => !answers['version'],
+            when: answers => answers['version'] == 'custom',
```
Comment (mjschock) [inline lib/release/questions.js]: does this still work if nothing is selected?

File: package.json [hunk 2]
```diff
@@ -98,6 +99,7 @@
     "babel-loader": "^6.2.4",
     "babel-preset-es2015": "^6.6.0",
     "bcapp-pattern-lab": "git@github.com:bigcommerce-labs/bcapp-pattern-lab.git",
+    "bower": "^1.8.2",
```
Comment (mjschock) [inline package.json]: do we need bower?

Comment (mattolson) [reply package.json]: Yes apparently ng-common depends on it but it is not defined as a dependency there, so `npm install` fails without this.

File: lib/release/questions.js (hunk 1 above)
Comment (mattolson) [reply lib/release/questions.js]: The only choices available for version are the well defined ones or `custom`

**Pull request merged**